torch/utils/data/graph.py

```diff
@@ -52,11 +52,18 @@ def reduce_hook(obj):
 
 
 def traverse(datapipe, only_datapipe=False):
+    cache: Set[IterDataPipe] = set()
+    return _traverse_helper(datapipe, only_datapipe, cache)
+
+
+# Add cache here to prevent infinite recursion on DataPipe
+def _traverse_helper(datapipe, only_datapipe=False, cache=None):
```

Comment:
Why does `cache` have a default here? There is not guard against `None` so not passing `cache` results in an exception. Given that this is a private helper, maybe we can simply remove the defaults?

```suggestion
def _traverse_helper(datapipe, only_datapipe, cache):
```

Reply:
makes sense.